### DISCUSSION TITLE
How to get cursor position and use it in a template?

### DISCUSSION THREAD

**Discussion by serpro69** [SilentVoid13/Templater#540] [Help]
Struggling to get the position of the cursor (indentation level) and then use that inside the template. 
Basically what I want to do is when a template is inserted not at first char position, then the whole template will be indented with the same level where the cursor was. 
Pretty sure I need some custom js code for this,  access the obsidian (or rather codemirror?) internal api , assign the cursor position to a variable? But I'm not that familiar with the above to be able to successfully piece it together myself it seems. So any help would be much appreciated.

**Comment by AB1908** [comment]
I think getting the cursor position isn't quite doable with templater. Would def need to interface with CodeMirror, if I understand correctly. I'll see if I can find something.

**Reply by AB1908** [reply]
Wait, can you show me an example? I'm not entirely certain we need the cursor position.

**Reply by AB1908** [reply]
I did ask @shabegom about this and they said:

> Yep, easiest way would be to get the codemirror Editor and use getCursor().

**Reply by serpro69** [reply]
> Wait, can you show me an example? I'm not entirely certain we need the cursor position.

I don't have access to my vault right now, but it's easily reproducible with any multi line template. 

1. Create a template with several lines of text
2. Insert a template at cursor position != 0
3. First line of the template is inserted in the indented cursor position, all other lines are inserted at the beginning of the line

**Reply by AB1908** [reply]
Hmm an alternative to using CodeMirror's API would be to fetch file contents, add some conditional logic and then reinsert with a certain indent. Doesn't sound too bad IMO.

**Reply by serpro69** [reply]
> Hmm an alternative to using CodeMirror's API would be to fetch file contents, add some conditional logic and then reinsert with a certain indent. Doesn't sound too bad IMO.

I think that could end up being very hacky, especially in cases where you insert a template not at the end of a file, but somewhere in the middle.

So far the workaround I'm using is to select the indentation level, and then use the `tp.file.selection` to indent each line in the template. This works, but still requires extra steps when inserting the template, which is not ideal. Hence getting the exact cursor position instead of using selection would be much better.

**Reply by AB1908** [reply]
Makes sense, try using shabegom's suggestion I mentioned above.
